src/peripheral/scb.rs

```diff
@@ -1011,4 +1013,96 @@ impl SCB {
             });
         }
     }
+
+    /// Enable the exception
+    ///
+    /// If the exception is enabled, when the exception is triggered, the exception handler will be executed instead of the
+    /// HardFault handler.
+    /// This function is only allowed on the following exceptions:
+    /// * `MemoryManagement`
+    /// * `BusFault`
+    /// * `UsageFault`
+    /// * `SecureFault` (can only be enabled from Secure state)
+    ///
+    /// Calling this function with any other exception will do nothing.
+    #[inline]
+    #[cfg(not(any(armv6m, armv8m_base)))]
+    pub fn enable(&mut self, exception: Exception) {
+        if self.is_enabled(exception) {
+            return;
+        }
+
+        // Make sure that the read-modify-write sequence happens during a critical section to avoid
+        // modifying pending and active interrupts.
+        interrupt::free(|_| {
+            let shift = match exception {
+                Exception::MemoryManagement => 16,
```

Comment:
If we made `enum Exception` C-style with values, we wouldn't need to copy/paste this lookup table three times.

Reply:
I can use the `enum SystemHandler` instead which carries more or less the same meaning and is a C-style enum with fixed value (the exception pritory)?
I would however still have to check that the `exception` given is only one of the 4 allowed in every function. I could use the fact that this enum starts at `MemoryManagement` and the four values are contiguous to only check the upper bound but what a new exception with a lower priority than `MemoryManagement` is added in the future 🤯 

I could also create a new private function that does this translation.

Reply:
A new private function or a private C style enum would work. I agree using types defined elsewhere is problematic.

Reply:
Added a private method on the last commit.